src/main/java/org/littleshoot/proxy/mitm/SubjectAlternativeNameHolder.java

```diff
@@ -47,8 +47,8 @@ public void addAll(Collection<List<?>> subjectAlternativeNames) {
     }
 
     private ASN1Encodable parseGeneralName(List<?> nameEntry) {
-        if (nameEntry != null && nameEntry.size() != 2) {
-            throw new IllegalArgumentException(String.valueOf(nameEntry));
+        if (nameEntry != null && nameEntry.size() != 2 || nameEntry == null) {
```

Comment:
I think this 'if' block could be split into two 'if' blocks - the first one for the null check and the second one for the old condition. Currently, I think it is a little hard-to-understand.